src/commonMain/kotlin/it/krzeminski/snakeyaml/engine/kmp/serializer/Serializer.kt

```diff
@@ -33,6 +34,10 @@ class Serializer(
 ) {
     private val serializedNodes: MutableSet<Node> = mutableSetOf()
     private val anchors: MutableMap<Node, Anchor?> = mutableMapOf()
+    private val isDereferenceAliases: Boolean = settings.isDereferenceAliases
+    // TODO: rewrite Collections.newSetFromMap(new IdentityHashMap<Node, Boolean>());
+    //   Regular set is incorrect!
+    private val recursive: MutableSet<Node> = mutableSetOf()
```

Comment:
⚠️ This has to be fixed before merging. From a quick skim, I haven't found any set in Kotlin that would do identity-based equality check, so I think we need to implement it ourselves.

Reply:
TODO: check if https://github.com/krzema12/snakeyaml-engine-kmp/blob/6c52b610382094f75c4d639d1d2f3fa40c68a4a1/src/commonMain/kotlin/it/krzeminski/snakeyaml/engine/kmp/representer/BaseRepresenter.kt#L225 can be a solution to this problem.

Reply:
I ended up implementing our own multiplatform `IdentitySet` based on utils that the KMP API delivers. It looks sane to me, but definitely is a subject of review here.